### PR TITLE
184752108 Buyer should be able to view their cart

### PR DESCRIPTION
**Pull request by ReneNtabana**
## What does this PR do?
A buyer should be able to view products on the cart
## Description of Task to be completed?
- The cart should be retrieved from the buyer's session
- The product details (name, price, image) should be retrieved from the database for each item in the cart
- The cart total should be calculated by summing the prices of the items in the cart
- The cart should be sent to the frontend to be displayed to the buyer
## How should this be manually tested?
- Clone the repository using `git clone` [link](https://github.com/atlp-rwanda/e-comm-team-techsmith-bn)

- Install node packages that the application depends on: `npm i`

- Checkout to the branch using `git checkout <branch-name>`

- Create a `.env` file in the root directory and add all environment variables required to run the application. The required variables are listed in the `.env.example` file available in root directory of the application.

- Start the application development server: `npm run dev`

## What are the relevant pivotal tracker/Trello stories?
[184752108](https://www.pivotaltracker.com/story/show/184752108)
## Screenshots (if appropriate)
Screenshot (227)

### CODE REVIEW

File: src/test/cart.test.js [hunk 1]
```diff
@@ -52,3 +52,28 @@ describe('Product already in cart',() => {
         });
   });
 });
+
+// describe('Viewing products on cart',() => {
```
Comment (niyontwali) [inline src/test/cart.test.js:56]: Remove commented codes if they are not needed, if those tests are needed uncomment them and refine

**Review by irakevine — COMMENTED** [review]
Cartresponse you should change it and use its own because it is retrieve data that's why you should create a new one in the component and name it as retrieveResponse and put everything it should have then it will bring your response you get when I do getAllCart.

**Comment by bergad37** [comment]
I suggest that you rename the class `addProductController`  to `cartController` for clarity so that other CRUD operations like `update cart` can be included in this class. Otherwise well done!

**Review by irakevine — COMMENTED** [review]
It's great that you changed it

**Review by nishimweprince — APPROVED** [review]
Excellent work!!

**Review by irakevine — APPROVED** [review]
You did a great job and I like how your code without repeating the same. thing makes sense

**Review by bergad37 — COMMENTED** [review]
I left a suggestion above!

**Review by bergad37 — APPROVED** [review]
Well done,!

**Review by Dyves1 — APPROVED** [review]
Good, this is really important to us

**Review by niyontwali — APPROVED** [review]

**Pull request merged**